appveyor.yml

```diff
@@ -19,4 +19,4 @@ clone_depth: 1
 build: false
 
 test_script:
-  - cargo test --features=deny-warnings
+  - cargo test --features=deny-warnings -- --ignored
```

Comment:
This seems like an extra change causing the failures?

Reply:
That change was intentional, to support "Add new test runner behaviour on Windows (it will ignore symlink-related tests and instead run them if you specify --ignored.)" because on Appveyor, tests are run with enough privileges.

However, there were some doctests that were using the `ignore` functionality to hide from the compiler. This has now been fixed.

Reply:
We sometimes intentionally mark tests as `#[ignore]`, so I don't think we can do this.

I recently added a support function [`symlink_supported`](https://github.com/rust-lang/cargo/blob/137a36f07004c33eaf742b85c04abe296aa4aa17/tests/testsuite/support/mod.rs#L1773-L1803).  Can you change the tests to [check it](https://github.com/rust-lang/cargo/blob/137a36f07004c33eaf742b85c04abe296aa4aa17/tests/testsuite/dep_info.rs#L481-L483)  instead of using the ignore attribute?

Reply:
> We sometimes intentionally mark tests as #[ignore], so I don't think we can do this.

I've tracked down those cases, and as far as I could figure out either they were fine with running on CI, or `ignore` was used to hide completely broken tests ([compile](https://github.com/rust-lang/cargo/pull/6817/files#diff-5909389e24cc2ed6babe2cd8f244a96eR6) [error](https://github.com/rust-lang/cargo/pull/6817/files#diff-f341aafa15818034baebb1b00be460c6R24) in [doctest examples](https://github.com/rust-lang/cargo/pull/6817/files#diff-f9445846d299388561a3972b309c9d9eR76), for example). I'd argue ``#[ignore]`` marked tests are supposed to be run at least at some times, because they should otherwise have been marked with ``cfg`` attributes.

I don't think ``#[ignore]`` is currently being used to hide long-running tests, so enabling ``--ignored`` should be fine. It may still be worthwhile to add ``symlink_supported``, though.

Unfortunately Rusts test runner doesn't support test skipping, because I'd really like to be able to indicate why a certain test is not being run instead of silently marking it as a pass.

Reply:
The intent with using `#[ignore]` (at least for me) is to add a test that is known to break, but you want to keep to ensure it continues to compile, and to provide some visibility that there is something unfinished.  You can't do that with `#[cfg]`, since it gets pruned before compiling.  I'm glad you fixed the existing things that weren't working, but in the future it would prevent us from using `#[ignore]`. 

I'd still prefer to remove the CI changes, and it looks like there's also a rustfmt error.

Reply:
Okay, but then this would probably be a good opportunity to revisit the two remaining `#[ignore]`s:

* [`tests/testsuite/version::version_works_without_rustc`](https://github.com/rust-lang/cargo/blob/master/tests/testsuite/version.rs#L17-L22)

Seems to work fine on Windows...

* [`tests/testsuite/concurrent::killing_cargo_releases_the_lock`](https://github.com/rust-lang/cargo/blob/master/tests/testsuite/concurrent.rs#L372-L435)

This test remarks that it doesn't work on older Windows versions, but it does work on Appveyors CI services – this test probably should be run most of the time.